src/js/views/patients/patient/sidebar/sidebar_views.js

```diff
@@ -1,142 +1,151 @@
+      },
+    }),
+  },
+};
 
-const GroupsView = View.extend({
-  behaviors: [
-    {
-      behaviorClass: ModelRender,
-      changeAttributes: ['_groups'],
-    },
-  ],
+const widgetView = View.extend({
   className: 'patient-sidebar__section',
-  template: PatientSidebarGroupsTemplate,
+  template: hbs`{{#if display_name}}<div class="patient-sidebar__heading">{{ display_name }}</div>{{/if}}<div class="patient-sidebar__item" data-content-region></div>`,
   templateContext() {
     return {
-      groups: _.map(this.model.getGroups().models, 'attributes'),
+      display_name: this.getOption('config').get('display_name'),
     };
   },
-});
-
-const EngagementStatusPreloadView = View.extend({
-  template: hbs`{{ @intl.patients.patient.sidebar.sidebarViews.engagementStatusPreloadView.loading }}`,
-  onRender() {
-    anime({
-      targets: this.el,
-      opacity: 0.5,
-      loop: true,
-      easing: 'easeInOutSine',
-      duration: 400,
-      direction: 'alternate',
-    });
+  regions: {
+    content: '[data-content-region]',
   },
-});
-
-const EngagementStatusView = View.extend({
-  className: 'patient-sidebar__engagement-status',
-  template: hbs`
-    {{#if engagement.status}}
-      <span class="engagement-status__icon {{ engagement.status }} u-margin--r-4">{{fas "circle"}}</span>{{formatMessage (intlGet "patients.patient.sidebar.sidebarViews.engagementStatusView.status") status=engagement.status}}
-    {{else}}
-      <span class="patient-sidebar__no-engagement">{{ @intl.patients.patient.sidebar.sidebarViews.engagementStatusView.notAvailable }}</span>
-    {{/if}}
-  `,
-  triggers: {
-    'click': 'click',
+  onRender() {
+    const view = this.getOption('view');
+    this.showChildView('content', new view(this.options));
   },
 });
 
-const SidebarView = View.extend({
+const SidebarView = CollectionView.extend({
   className: 'patient-sidebar flex-region',
-  template: PatientSidebarTemplate,
-  templateContext() {
-    return {
-      engagementEnabled: this.getOption('orgEngagementEnabled'),
-    };
+  template: false,
+  onRender() {
+    this.renderWidgets();
   },
-  regionClass: Region.extend({ replaceElement: true }),
-  regions: {
-    info: '[data-info-region]',
-    groups: '[data-groups-region]',
-    engagement: '[data-engagement-region]',
+  renderWidgets() {
+    const widgetConfigs = this.getOption('widgets');
+
+    widgetConfigs.each(this.renderWidget.bind(this));
   },
-  onRender() {
-    const model = this.model;
+  renderWidget(config) {
+    const widget = sidebarWidgets[config.get('widget_type')];
+
+    /* istanbul ignore next: prevents widget render error, remove when widgets are no longer hardcoded in app */
+    if (!widget) return;
 
-    this.showChildView('info', new InfoView({ model }));
-    this.showChildView('groups', new GroupsView({ model }));
+    this.addChildView(new widgetView({
+      view: widget.view,
+      config,
```

Comment:
I think the model should be config and `patient: this.model` might be better for the `widgetView` itself.. then that view can pass the patient as the model to the widget